### PR TITLE
Update switch user to logout and login for s390x

### PR DESCRIPTION
**Pull request by lemon-suse**
From bug https://bugzilla.suse.com/show_bug.cgi?id=1185943, 'switch user' by button is not supported on s390x VM and we need to logout and login new user for the switch user test.

- Related ticket: https://progress.opensuse.org/issues/92284
- Needles: N/A
- Verification run: https://openqa.nue.suse.com/tests/6093177#step/check_upgraded_service/17
                          http://openqa.nue.suse.com/tests/6093200#    desktop test run avoid regression

### CODE REVIEW

**Comment by zhaocongjia** [comment]
LGTM, this change shall fix the issue encountered on s390x.

File: lib/services/users.pm [hunk 1]
```diff
@@ -43,13 +43,16 @@ sub lock_screen {
...
 sub logout_and_login {
+    my ($login_user, $login_pw) = @_;
+    my $test_user = $login_user // $username;
+    my $test_pw   = $login_pw   // $newpwd;
     handle_logout;
     send_key_until_needlematch 'displaymanager', 'esc', 9, 10;
     mouse_hide();
     wait_still_screen;
-    assert_and_click "displaymanager-$username";
+    assert_and_click "displaymanager-$test_user";
```
Comment (tinawang123) [inline lib/services/users.pm:53]: Do you need add new needle at here?

Comment (lemon-suse) [reply lib/services/users.pm:53]: No, the needle displaymanager-test already in openQA.

File: lib/services/users.pm [hunk 2]
```diff
@@ -165,7 +168,12 @@ sub full_users_check {
     }
     else {
         #swtich to new added user then switch back
-        switch_users;
+        if (check_var('ARCH', 's390x')) {
```
Comment (coolgw) [inline lib/services/users.pm:171]: more comments need why we handle s390 different

Comment (rfan1) [inline lib/services/users.pm:171]: Any known issues/bugs we can't switch user like other platforms? seems we add a workaround here

Comment (lemon-suse) [reply lib/services/users.pm:171]: See PR description, comments#2 of https://bugzilla.suse.com/show_bug.cgi?id=1185943

Comment (lemon-suse) [reply lib/services/users.pm:171]: See PR description, comments#2 of https://bugzilla.suse.com/show_bug.cgi?id=1185943

Comment (rfan1) [reply lib/services/users.pm:171]: Yup, got it.

**Review by rfan1 — APPROVED** [review]
LGTM

File: lib/services/users.pm (hunk 2 above)
Comment (coolgw) [reply lib/services/users.pm:171]: I mean put comment into the code, not the PR

Comment (lemon-suse) [reply lib/services/users.pm:173]: Update code with the comments for the reason.

**Review by tinawang123 — APPROVED** [review]
LGTM

**Pull request merged**